src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1924,14 +1936,22 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
 
     buildInternalRegisterUses();
 
-    if ((dstCount == 1) || (dstCount == 2))
+    if ((dstCount == 1) || (dstCount == 2) || (dstCount == 3) || (dstCount == 4))
     {
         BuildDef(intrinsicTree, dstCandidates);
 
-        if (dstCount == 2)
```

Comment:
I'm curious as to why this wasn't required for `AdvSimd.LoadVector123x2/3/4`

Reply:
I'm curious why as well.

Reply:
Ok, it turns out we don't need change this part. I think in the beginning when I was implementing this, I had to make those changes, but since I added the NeedsConsectiveRegisters flag we do not need to.